src/libraries/System.Runtime/tests/System.Runtime.Tests.csproj

```diff
@@ -332,6 +333,26 @@
     <TrimmerRootDescriptor Include="$(MSBuildThisFileDirectory)ILLink.Descriptors.xml" />
   </ItemGroup>
 
+  <!--
+      These tests are currently incompatible with TestReadyToRun.
+      We have a tracking issue herehttps://github.com/dotnet/runtime/issues/94189
+  -->
+  <ItemGroup Condition="'$(TestReadyToRun)' == 'true'">
+    <Compile Remove="System\Type\TypeTests.Get.cs" />
+    <Compile Remove="System\Runtime\JitInfoTests.cs" />
+    <Compile Remove="System\EnumTests.cs" />
+  </ItemGroup>
+
+  <!--
+      Timezone tests are not working on Linux on CI when TestReadyToRun is enabled.
+      It might be something missing with the machines. We have a tracking issue here:
+      https://github.com/dotnet/runtime/issues/94190
+  -->
+  <ItemGroup Condition="'$(TargetOS)' == 'linux' and '$(TestReadyToRun)' == 'true'">
+    <Compile Remove="System\TimeZoneInfoTests.cs" />
+    <Compile Remove="$(CommonTestPath)System\TimeProviderTests.cs" />
+  </ItemGroup>
+
```

Comment:
This is not how we do test exclusions in this repo.

We exclude failing tests with an `ActiveIssue` attribute. For example:

https://github.com/dotnet/runtime/blob/848c729ca44f3ae651ecd4672d40111508770450/src/mono/wasm/debugger/DebuggerTestSuite/SetNextIpTests.cs#L116

We need some way to detect the test was ReadyToRun compiled so that the ActiveIssue can be conditioned on that.

I would add a `PlatformDetection.IsReadyToRunCompiled` here: https://github.com/dotnet/runtime/blob/848c729ca44f3ae651ecd4672d40111508770450/src/libraries/Common/tests/TestUtilities/System/PlatformDetection.cs

As for the implementation, I don't know if we have a good API to detect this, but in the worst case, one could set an environment variable in the single file test runner and check for that:

* Add a new `#define` constant somewhere around here that is set for TestReadyToRun only: https://github.com/dotnet/runtime/blob/848c729ca44f3ae651ecd4672d40111508770450/eng/testing/tests.singlefile.targets#L66-L68
* Add a SetEnvironmentVariable call (under the above defined `#ifdef`) to somewhere around here: https://github.com/dotnet/runtime/blob/848c729ca44f3ae651ecd4672d40111508770450/src/libraries/Common/tests/SingleFileTestRunner/SingleFileTestRunner.cs#L30-L31
* Implement `PlatformDetection.IsReadyToRunCompiled` by reading the environment variable.
* Block the failing tests with `[ActiveIssue("https://...", typeof(PlatformDetection), nameof(PlatformDetection.IsReadyToRunCompiled))]`

Reply:
Thank you very much for the detailed explanation Michal. I didn't know about this way of excluding tests, so I just went the route that we have in coreclr with `issues.targets`. I'm not sure we have a way of detecting the R2R-compilation of the tests, so I will take your suggestion of the environment variable. I will also then file a tracking issue to add this functionality.